### ISSUE TITLE
Fail to install chart when "Legacy authorization" disabled on GCE

### ISSUE THREAD

**Issue by saulshanabrook** [jupyterhub/zero-to-jupyterhub-k8s#118]
The pod fails to start when I have "Legacy Authorization" disabled in Google Container Engine:

```
2017-07-14T16:15:04.059433146Z Error from server (Forbidden): the server does not allow access to the requested resource (get nodes)
2017-07-14T16:15:04.06317655Z Pulling jupyterhub/k8s-singleuser-sample:v0.4 on 0 nodes
2017-07-14T16:15:04.316576363Z Error from server (Forbidden): error when retrieving current configuration of:
2017-07-14T16:15:04.316612457Z &{0xc4205f8900 0xc420258540 default pull--v0-4-1500048904 STDIN &Job{ObjectMeta:k8s_io_kubernetes_pkg_api_v1.ObjectMeta{Name:pull--v0-4-1500048904,GenerateName:,Namespace:,SelfLink:,UID:,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{},Annotations:map[string]string{kubectl.kubernetes.io/last-applied-configuration: ,},OwnerReferences:[],Finalizers:[],ClusterName:,},Spec:JobSpec{Parallelism:*0,Completions:*0,ActiveDeadlineSeconds:*1200,Selector:nil,ManualSelector:nil,Template:k8s_io_kubernetes_pkg_api_v1.PodTemplateSpec{ObjectMeta:ObjectMeta{Name:,GenerateName:,Namespace:,SelfLink:,UID:,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{yuvi.in/pod-puller: true,},Annotations:map[string]string{scheduler.alpha.kubernetes.io/affinity: {
2017-07-14T16:15:04.316627211Z   "podAntiAffinity": {
2017-07-14T16:15:04.31663287Z     "requiredDuringSchedulingIgnoredDuringExecution": [{
2017-07-14T16:15:04.316637067Z        "topologyKey": "kubernetes.io/hostname",
2017-07-14T16:15:04.316641721Z        "labelSelector": {
2017-07-14T16:15:04.316646437Z          "matchLabels": { "yuvi.in/pod-puller": "true"}
2017-07-14T16:15:04.316652013Z        }
2017-07-14T16:15:04.316655997Z     }]
2017-07-14T16:15:04.316659964Z   }
2017-07-14T16:15:04.31666451Z }
2017-07-14T16:15:04.316676824Z ,},OwnerReferences:[],Finalizers:[],ClusterName:,},Spec:PodSpec{Volumes:[{docker-socket {HostPathVolumeSource{Path:/var/run/docker.sock,} nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil}}],Containers:[{puller yuvipanda/image-puller:v0.2 [] [jupyterhub/k8s-singleuser-sample v0.4]  [] [] {map[] map[]} [{docker-socket false /var/run/docker.sock }] nil nil nil   nil false false false}],RestartPolicy:Never,TerminationGracePeriodSeconds:nil,ActiveDeadlineSeconds:nil,DNSPolicy:,NodeSelector:map[string]string{},ServiceAccountName:,DeprecatedServiceAccount:,NodeName:,HostNetwork:false,HostPID:false,HostIPC:false,SecurityContext:nil,ImagePullSecrets:[],Hostname:,Subdomain:,},},},Status:JobStatus{Conditions:[],StartTime:<nil>,CompletionTime:<nil>,Active:0,Succeeded:0,Failed:0,},} &TypeMeta{Kind:,APIVersion:,}  false}
2017-07-14T16:15:04.316686233Z from server for: "STDIN": the server does not allow access to the requested resource (get jobs.batch pull--v0-4-1500048904)
```

Once I disabled this, it worked. Is it possible to support [RBAC](https://cloud.google.com/container-engine/docs/role-based-access-control) with this project? I have no knowledge on the differences, just the label "legacy" made me want to turn off that option.

**Comment by yuvipanda** [comment]
+1, 0.4.1 should have this!

**Comment by yuvipanda** [comment]
I took a shot at this, and unfortunately it looks like v0.4.1 will not have this - primarily due to the fact that our image pre-puller is a total hack and we can't fit a role binding into helm in the location we want to.

I did commit actual RBAC rules to master however, so everything except the pre-puller should work. Since the pre-puller should be replaced with a custom controller soon, this is probably ok.
